### ISSUE TITLE
About the lint behavior of scheduled job

### ISSUE THREAD

**Issue by kenchan0130** [rhysd/actionlint#14]
When I run `actionlint`, the output is as follows:

```log
.github/workflows/schedule.yml:4:13: scheduled job runs too frequently. it runs once per 60 seconds [events]
  |
4 |     - cron: '* * * * *'
  |             ^~
```

According to the documentation at https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events, it seems that '* * * * *' appears to be the correct format.

Which is correct, GitHub Actions document or `actionlint` implementation?

**Comment by rhysd** [comment]
Yes, it's correct format, but it is executed too frequently. The workflow will be repeatedly run without any interval and will cause some issues (will hit some resource restriction of GitHub Actions, I believe). It's mostly mistake so actionlint reports it as an error.

**Comment by rhysd** [comment]
Since running a workflow generally takes more than one minute, it will hit this limitation.

> Workflow run queue - No more than 100 workflow runs can be queued in a 10 second interval per repository. If a workflow run reaches this limit, the workflow run is terminated and fails to complete.

https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration

I think you would need to consider reducing the frequency like once per hour or once per day.

**Comment by rhysd** [comment]
I found a evidence in community forum. GitHub staff said:

> The minimum amount of time you can schedule a workflow to run is every 5 minutes. Because of this pooling you may see runs being “skipped” if there are already runs waiting in the pool by the time your next schedule is set to run.

https://github.community/t/schedule-every-5-mins-but-runs-a-bit-randomly/159355/4

So your configuration is correct as format, but frequency is too high as GitHub Actions configuration.

**Comment by kenchan0130** [comment]
I understand what you are trying to say.

If I set the interval to every 2 minutes, like */2* * *, the lint will exit successfully.
So am we correct in assuming that this lint is asking you to set the interval to more than 2 minutes?

If it is https://github.com/rhysd/actionlint/issues/14#issuecomment-886662264, I think a better output message would be to allow an interval of **at least 5 minutes**.

**Comment by shogo82148** [comment]
The change log is here:

> https://github.blog/changelog/2019-11-01-github-actions-scheduled-jobs-maximum-frequency-is-changing/
> In order to provide a more reliable experience, we’ve set the minimum time period for scheduled runs to 5 minutes. A schedule of `* * * * *` which would previously run every minute will now run every five minutes.

And the document says that the shortest interval is once every 5 minutes.

> https://docs.github.com/en/actions/reference/events-that-trigger-workflows#:~:text=The%20shortest%20interval%20you%20can%20run%20scheduled%20workflows%20is%20once%20every%205%20minutes.
> The shortest interval you can run scheduled workflows is once every 5 minutes.

**Comment by rhysd** [comment]
> So am we correct in assuming that this lint is asking you to set the interval to more than 2 minutes?

Yes, actionlint asks you to make the interval more than 2 min. Since I didn't know exact duration (5 min) when I implemented the check, I set the interval threshold to 1 min.

> I think a better output message would be to allow an interval of at least 5 minutes.

Yeah, that might be more exact. The intension of the interval check is that I wanted to catch the mistake like `* */5 * * *` (actually some famous repository did the mistake).

**Comment by rhysd** [comment]
@shogo82148 Thank you for your follow up with the more concrete description.

**Comment by rhysd** [comment]
I will change the interval check from 1 min to 5 min.

**Comment by shogo82148** [comment]
I'm wondering that https://github.blog/changelog/2019-11-01-github-actions-scheduled-jobs-maximum-frequency-is-changing/ is not listed in https://github.blog/changelog/label/actions/ 🤔 
It took a long time to find this article...

**Comment by rhysd** [comment]
Umm, I guess it is a kind of mistake because the article is obviously for GitHub Actions. Thank you for taking time to find it.

**Comment by rhysd** [comment]
I changed minimum frequency to once per every 5 min at 04696da.

**Comment by kenchan0130** [comment]
Thank you for your confirmation and changes!

**Comment by shogo82148** [comment]
It looks that https://github.com/rhysd/actionlint/commit/04696daa81b874dc8b4df51f509ec7f61133027d warns `*/5 * * * *`, however it's OK because the interval can be every 5 minutes.

```go
	// (#14) https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events
	//
	// > The shortest interval you can run scheduled workflows is once every 5 minutes.
	if diff <= 60.0*5 {
		rule.errorf(spec.Pos, "scheduled job runs too frequently. it runs once per %g seconds. the shortest interval is once every 5 minutes", diff)
	}
```

It should be:

```go
	// (#14) https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events
	//
	// > The shortest interval you can run scheduled workflows is once every 5 minutes.
	if diff < 60.0*5 {
		rule.errorf(spec.Pos, "scheduled job runs too frequently. it runs once per %g seconds. the shortest interval is once every 5 minutes", diff)
	}
```
